### DISCUSSION TITLE
Keycloack add certificates to truststore for outgoing connection

### DISCUSSION THREAD

**Discussion by vivek862** [keycloak/keycloak#15265] [Archive]
Hi ,

We are looking for adding certficaes to the keycloack truststore to secure outgoing connection dynamically . the supported method for configuring a truststore for external TLS as provided [here](https://www.keycloak.org/server/keycloak-truststore) is as follows:

bin/kc.[sh|bat] start --spi-truststore-file-file=myTrustStore.jks --spi-truststore-file-password=password --spi-truststore-file-hostname-verification-policy=ANY

I am working on a use case where all the external certs are not known beforehand. An example could be adding a new IDP provider on a running keycloack instance.
Is there a a runtime configuration possible of such external certificates ?

**Comment by sschu** [comment]
You mean your external partners use self-signed certificates that are not known beforehand? I don't think this will work. But with certificates signed by trusted CAs, all the CA certificates should already be be available.
